ddp-workspace/projects/ddp-dsm-ui/src/app/file-download/file-download.component.ts

```diff
@@ -1,67 +1,68 @@
 import {DatePipe} from '@angular/common';
-import {Component, Input, OnDestroy, Output} from '@angular/core';
-import {File} from '../participant-list/models/file.model';
+import {Component, Input} from '@angular/core';
+import {mergeMap} from 'rxjs';
+import {take} from 'rxjs/operators';
+import {ESFile} from '../participant-list/models/file.model';
 import {Participant} from '../participant-list/participant-list.model';
 import {ComponentService} from '../services/component.service';
 import {DSMService} from '../services/dsm.service';
 import {Utils} from '../utils/utils';
-import { Subscription } from 'rxjs';
 
 const fileSaver = require( 'file-saver' );
 
+
 @Component( {
   selector: 'app-file-download',
   templateUrl: './file-download.component.html',
   styleUrls: [ './file-download.component.scss' ]
 } )
 
-export class FileDownloadComponent implements OnDestroy {
+export class FileDownloadComponent {
   @Input() participant: Participant;
   downloadMessage = '';
   downloading = false;
   CLEAN = 'CLEAN';
-  subscription: Subscription;
+  private SUCCESSFUL_DOWNLOAD_MESSAGE = 'File download finished.';
+  private NOT_SCANNED_FILE_MESSAGE = 'Error - file has not passed scanning';
 
-  constructor(private dsmService: DSMService ) {
+  constructor( private dsmService: DSMService ) {
 
   }
 
-  ngOnDestroy(): void {
-    if (this.subscription != null) {
-      this.subscription.unsubscribe();
-    }
-  }
 
-  downloadParticipantFile( file: File ): void {
+  downloadParticipantFile( file: ESFile ): void {
     this.downloading = true;
-    if (file.scanResult !== this.CLEAN) {
-      this.downloadMessage = 'Error - file has not passed scanning';
-      this.downloading = false;
+    if (!this.isFileClean( file )) {
+      this.setDownloadMessageAndStatus( this.NOT_SCANNED_FILE_MESSAGE, false );
       return;
     }
+
     const realm = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
-    this.subscription = this.dsmService.downloadParticipantFile( file.fileName, file.bucket, file.blobName, realm, file.mimeType )
-      .subscribe( {
+    this.dsmService.getSignedUrl( this.participant.data.profile[ 'guid' ], file, realm ).pipe(
+      mergeMap(data => this.dsmService.downloadFromSignedUrl(data['url']).pipe(take(1)))).subscribe( {
```

Comment:
I think is important to handle the subscription in order to make sure it is going to be unsubscribed after the component is destroyed. I suggest the following:

this.subscription = this.dsmService.getSignedUrl( this.participant.data.profile[ 'guid' ], file, realm ).pipe(
      mergeMap(data => this.dsmService.downloadFromSignedUrl(data['url']).pipe(take(1)))).subscribe(...)

Reply:
Right. I don't think think that second .pipe() method is necessary.
I think the whole thing should look something like:
```typescript
this.dsmService.getSignedUrl(blah,blah).pipe(
        mergeMap(data => this dsmService.downloadFromSignedUrl(blah,blah)),
        take(1)).subscribe(blah,blah,blah);
```

Reply:
That's ok , but my point is that we need to unsubscribe from the observavble once the component gets destroyed . That's what I think is still missing.

Reply:
@alextanlan Please correct me if I'm wrong.

Reply:
@MocanaAtBroad @juanseba21 I am sorry I missed these comments before merging my branch. I will make a branch for the fixes you requested

Reply:
@MocanaAtBroad when writing it like `mergeMap(data => this.dsmService.downloadFromSignedUrl(data['url']), take(1)))` I get the error and can not build ` Type 'Observable<unknown>' is not assignable to type 'BlobPart'.
  Type 'Observable<unknown>' is missing the following properties from type 'ArrayBufferView': buffer, byteLength, byteOffset`

making it go through a pipe keeps the  Observable<ArrayBuffer> type

Reply:
@juanseba21 adding the `take(1)` takes care of unsubscribing here, also HTTPClient takes care of unsubscribing itself

Reply:
I agree that [Marco's comment](https://github.com/broadinstitute/ddp-angular/pull/1638#discussion_r942726640) is valid here. 
Using `take(1)` is enough for our case. 
So it will look like:

```
this.dsmService.getSignedUrl(this.participant.data.profile['guid'], file, realm)
    .pipe(
        mergeMap(data => this.dsmService.downloadFromSignedUrl(data['url'])),
        take(1)
    )
    .subscribe(...)
```
It will unsusbcribe after the first use.

Reply:
As for the [Juan comment](https://github.com/broadinstitute/ddp-angular/pull/1638#discussion_r942543769), I suppose it is about the (rare) case when the component is destroyed but the subscription is still is not completed (like [an example](https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f#:~:text=Beware%2C%20even%20if%20the%20AppComponent%20is%20destroyed%20thesubscription%24%20will%20not%20unsubscribe%20until%20the%20interval%20emits%20a%20value.))

I believe in the particular case it is not important (and may be even desired  - we wait for for the file gets downloaded and then unsubscribe even though the component is (unlikely) destroyed for some reason)

Reply:
As an option/ a compromise we can use `takeUntil` operator instead of `take(1)` (see lots of its usages in our code).
Even it is overly here, imo.